packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -321,6 +326,75 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         );
     }
 
+    drawD3Chart(): JSX.Element {
+        const chartHeight = parseInt(this.props.style.height.toString());
+        const chartWidth = this.getChartWidth();
+
+        if (this.state.xyData.labels?.length > 0) {
+            const data: any[] = [];
+
+            this.state.xyData?.datasets?.forEach((dSet: any) => {
+                const row: any = [];
+                if (this.isScatterPlot) {
+                    dSet.data.forEach((tupple: any) => {
+                        row.push({ xValue: tupple.x, yValue: tupple.y });
+                    });
+                } else {
+                    dSet.data.forEach((y: number, j: number) => {
+                        row.push({ xValue: this.state.xyData.labels[j], yValue: y });
+                    });
+                }
+                data.push(row);
+            });
+
+            const yScale = scaleLinear()
+                .domain([this.state.allMin, Math.max(this.state.allMax, 1)])
+                .range([chartHeight - this.margin.bottom, this.margin.top]);
+
+            const xDomain = this.state.xyData.labels.length - 1;
+
+            const xScale = scaleLinear()
+                .domain([this.state.xyData.labels[0], this.state.xyData.labels[xDomain]].map(Number))
```

Comment:
The xScale is probably also incorrect due to the map(Number) of the bigint value causing loss of precision.

Reply:
Indeed. Using `getXForTime()` now.